`Full error: 'pydantic_core._pydantic_core.ValidationInfo' object has no attribute 'manifest'.`

### Have you ensured that all of these are up to date?

- [x] Ape
- [x] Ape plugins

### What version of Ape are you on?

`0.8.38`

### What command(s) is the bug in?

`ape compile`

### Operating System

Linux

### Ape Configuration

### `ape-config.yaml`

```yaml
name: snekmate
contracts_folder: src/snekmate
default_ecosystem: ethereum
plugins:
  - name: vyper
vyper:
  evm_version: prague
compile:
  exclude:
    - r"(?!.*_mock\.vy$)"
```

### Description

As you might know, I run nightly tests in snekmate (against the latest `master` Vyper commit), and this includes `ape compile` tests: https://github.com/pcaversaccio/snekmate/blob/b0a0fde69b47bd9eeecb19da86d41c3aaca4ecb8/.github/workflows/test-contracts.yml#L48-L54

This night it started to fail with the following error (see [here](https://github.com/pcaversaccio/snekmate/actions/runs/18333106216/job/52220141602#step:7:1)):

```
Run ape compile
  ape compile
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.13.7/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.13.7/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.7/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.7/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.13.7/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.13.7/x64/lib
ERROR:    __local__.json manifest corrupted! Re-building.
Full error: 'pydantic_core._pydantic_core.ValidationInfo' object has no attribute 'manifest'.
WARNING:  Unable to load CLI endpoint for plugin 'ape_compile'
	ApeAttributeError: registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - registered_compilers - maximum recursion depth exceeded
	(Use `--verbosity DEBUG` to see full stack-trace)
Usage: ape [OPTIONS] COMMAND [ARGS]...
Try 'ape -h' for help.

Error: No such command 'compile'.
Error: Process completed with exit code 2.
```

To be clear, I haven't changed anything in the CI test flow for months and also use `ApeWorX/github-action@v3` for the setup. My best guess after very fast inspection is due to the recent Pydantic [`v2.12`](https://github.com/pydantic/pydantic/releases/tag/v2.12.0) release: https://pydantic.dev/articles/pydantic-v2-12-release, which gets installed with Ape: https://github.com/ApeWorX/ape/blob/885cb13e4bbc397a2bbfaa36df1ac49d3d7f8bba/setup.py#L103.

### Reproduction code

Fork [snekmate](https://github.com/pcaversaccio/snekmate) and run the CI workflow [🕵️‍♂️ Test smart contracts](https://github.com/pcaversaccio/snekmate/blob/main/.github/workflows/test-contracts.yml).

--- a/setup.py
+++ b/setup.py
@@ -100,6 +100,7 @@ setup(
         "packaging>=23.0,<24",
         "pandas>=2.2.2,<3",
         "pluggy>=1.3,<2",
+        # TODO: 2.12 introduces a breaking change (see #2701)
         "pydantic>=2.10.0,<2.12.0",
         "pydantic-settings>=2.5.2,<3",
         "pytest>=8.0,<9.0",

